python/ray/serve/handle.py

```diff
@@ -175,20 +181,22 @@ def _options(
         method_name: Union[str, DEFAULT] = DEFAULT.VALUE,
         multiplexed_model_id: Union[str, DEFAULT] = DEFAULT.VALUE,
         stream: Union[bool, DEFAULT] = DEFAULT.VALUE,
+        _router_cls: Union[str, DEFAULT] = DEFAULT.VALUE,
     ):
         new_handle_options = self.handle_options.copy_and_update(
             method_name=method_name,
             multiplexed_model_id=multiplexed_model_id,
             stream=stream,
+            _router_cls=_router_cls,
         )
 
-        if self._router is None:
+        if self._router is None and not _router_cls:
```

Comment:
Alternatively, I could move this to construct the router in the handle constructor. However, I didn't know if this deferred construction was intentional.